plugins/parsers/csv/parser.go

```diff
@@ -46,6 +54,65 @@ type Parser struct {
 	Log telegraf.Logger
 }
 
+type MetadataPattern []string
+
+func (record MetadataPattern) Len() int {
+	return len(record)
+}
+func (record MetadataPattern) Swap(i, j int) {
+	record[i], record[j] = record[j], record[i]
+}
+func (record MetadataPattern) Less(i, j int) bool {
+	// Metadata with longer lengths should be ordered before shorter metadata
+	return len(record[i]) > len(record[j])
+}
```

Comment:
I'm wondering if there are no already existing string array types that do provide this functionality.

Reply:
@srebhan do you know this?

Reply:
There is none that orders strings by their length AFAIK.